### PR TITLE
Change flags in GDT entries, add macro to print from pmode

### PR DESCRIPTION
**Pull request by sperezglz**
and add make rule to generate disassembly

Signed-off-by: Perez Gonzalez, Sergio <sperezglz@gmail.com>

### CODE REVIEW

File: boot/Makefile [hunk 1]
```diff
@@ -10,3 +10,6 @@ run: file.img
 	qemu-system-i386 -hda file.img
 clean:
 	rm boot.o file.img
+debug: boot.o
+	ld -o file.elf -T boot.ld boot.o; objdump -lS file.elf > file.lst
+
```
Comment (josuedhg) [inline boot/Makefile:15]: This is ok, but please add the created files to clean target

File: boot/boot.S [hunk 2]
```diff
@@ -39,8 +43,8 @@ halt:
 .p2align 2
 gdt:
 	.quad 0x0000000000000000 // null descriptor
-	.quad 0x00009a000000ffff // code descriptor type=execute/read
-	.quad 0x000092000000ffff // data descriptor type=read/write
+	.quad 0x00cf9a000000ffff // code descriptor type=execute/read
```
Comment (josuedhg) [inline boot/boot.S:43]: I am pretty sure there is a reason for this, please add it to the commit

**Comment by josuedhg** [comment]
Thank you for your collaboration :)

File: boot/boot.S (hunk 2 above)
Comment (aduejazz) [inline boot/boot.S:43]: Can you add which flags you added?

File: boot/boot.S [hunk 3]
```diff
@@ -39,8 +43,8 @@ halt:
...
+	.quad 0x00cf92000000ffff // data descriptor type=read/write
```
Comment (aduejazz) [inline boot/boot.S:44]: Same here (with the flags) ...

**Pull request merged**